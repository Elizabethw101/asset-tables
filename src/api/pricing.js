import { prices } from './mockData/prices'

export const getPricing = () => {
    return prices;
}