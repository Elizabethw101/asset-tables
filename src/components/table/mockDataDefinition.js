
 export const mockDataDefinition = [
        {
          label: "Ticker",
          dataType: "text",
          sortType: "alpha",
          dataField: "ticker"
        },
        {
          label: "Price",
          dataType: "currency",
          sortType: "number",
          dataField: "price"
        },
        {
          label: "Asset Class",
          dataType: "asset",
          sortType: "assetClass",
          dataField: "assetClass"
        }
      ];
