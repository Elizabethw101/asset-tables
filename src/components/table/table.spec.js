/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import Table from "./table";
import { getPricing } from "../../api/pricing";
import {mockDataDefinition} from './mockDataDefinition';

describe("it tests the table component", () => {
  let container = null;
  const pricing = getPricing();

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    container = null;
  });

  it("loads and renders the table", () => {
    container = renderer.create(
      <Table tableData={pricing} columnDefinition={mockDataDefinition} />
    );
    expect(container.toJSON()).toMatchSnapshot();
  });

  it("it expects has the right props", () => {
    container = mount(
      <Table tableData={pricing} columnDefinition={mockDataDefinition} />
    );
    expect(container.props().columnDefinition).toEqual(mockDataDefinition);
    expect(container.props().tableData).toEqual(pricing);
  });

  it("it expects table rows to match the data length plus the header", () => {
    container = mount(
      <Table tableData={pricing} columnDefinition={mockDataDefinition} />
    );
    expect(container.props().columnDefinition).toEqual(mockDataDefinition);
    expect(container.props().tableData).toEqual(pricing);

    const table = container.find('table');
    const rows = table.find('tr');
    expect(rows).toHaveLength(pricing.length +1);
  });

  it("it expects table to have three columns", () => {
    container = mount(
      <Table tableData={pricing} columnDefinition={mockDataDefinition} />
    );
    const table = container.find('table');
    const columns = table.find('th');
    expect(columns).toHaveLength(mockDataDefinition.length);
  });

  it("it expects the first header Labels text to be correct", () => {
    container = mount(
      <Table tableData={pricing} columnDefinition={mockDataDefinition} />
    );
    const table = container.find('table');
    const firstHeader = table.find('th').first();
    expect(firstHeader.text()).toBe('Ticker');
  });

});


