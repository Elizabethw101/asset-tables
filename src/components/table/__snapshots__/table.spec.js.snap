// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it tests the table component loads and renders the table 1`] = `
<table
  aria-label=""
  aria-rowcount={18}
  className="container"
  data-testid="data-table"
  role="table"
>
  <caption>
    
  </caption>
  <thead>
    <tr
      className="header"
    >
      <th
        className="hcell"
        onClick={[Function]}
      >
        Ticker
      </th>
      <th
        className="hcell"
        onClick={[Function]}
      >
        Price
      </th>
      <th
        className="hcell"
        onClick={[Function]}
      >
        Asset Class
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="cell credit"
      scope="row"
    >
      <td
        className="cell"
      >
        ALPHA
         
      </td>
      <td
        className="cell positive"
      >
        3150.67
         
      </td>
      <td
        className="cell"
      >
        Credit
         
      </td>
    </tr>
    <tr
      className="cell equities"
      scope="row"
    >
      <td
        className="cell"
      >
        BETA
         
      </td>
      <td
        className="cell positive"
      >
        3791.37
         
      </td>
      <td
        className="cell"
      >
        Equities
         
      </td>
    </tr>
    <tr
      className="cell equities"
      scope="row"
    >
      <td
        className="cell"
      >
        GAMMA
         
      </td>
      <td
        className="cell positive"
      >
        2299.1
         
      </td>
      <td
        className="cell"
      >
        Equities
         
      </td>
    </tr>
    <tr
      className="cell equities"
      scope="row"
    >
      <td
        className="cell"
      >
        DELTA
         
      </td>
      <td
        className="cell positive"
      >
        3132.66
         
      </td>
      <td
        className="cell"
      >
        Equities
         
      </td>
    </tr>
    <tr
      className="cell credit"
      scope="row"
    >
      <td
        className="cell"
      >
        EPSILON
         
      </td>
      <td
        className="cell positive"
      >
        1168.46
         
      </td>
      <td
        className="cell"
      >
        Credit
         
      </td>
    </tr>
    <tr
      className="cell credit"
      scope="row"
    >
      <td
        className="cell"
      >
        ZETA
         
      </td>
      <td
        className="cell positive"
      >
        2716.78
         
      </td>
      <td
        className="cell"
      >
        Credit
         
      </td>
    </tr>
    <tr
      className="cell macro"
      scope="row"
    >
      <td
        className="cell"
      >
        ETA
         
      </td>
      <td
        className="cell negative"
      >
        -3089.2
         
      </td>
      <td
        className="cell"
      >
        Macro
         
      </td>
    </tr>
    <tr
      className="cell macro"
      scope="row"
    >
      <td
        className="cell"
      >
        THETA
         
      </td>
      <td
        className="cell positive"
      >
        1075.44
         
      </td>
      <td
        className="cell"
      >
        Macro
         
      </td>
    </tr>
    <tr
      className="cell macro"
      scope="row"
    >
      <td
        className="cell"
      >
        IOTA
         
      </td>
      <td
        className="cell positive"
      >
        1096.64
         
      </td>
      <td
        className="cell"
      >
        Macro
         
      </td>
    </tr>
    <tr
      className="cell credit"
      scope="row"
    >
      <td
        className="cell"
      >
        KAPPA
         
      </td>
      <td
        className="cell positive"
      >
        2321.17
         
      </td>
      <td
        className="cell"
      >
        Credit
         
      </td>
    </tr>
    <tr
      className="cell credit"
      scope="row"
    >
      <td
        className="cell"
      >
        LAMBDA
         
      </td>
      <td
        className="cell positive"
      >
        1472.2
         
      </td>
      <td
        className="cell"
      >
        Credit
         
      </td>
    </tr>
    <tr
      className="cell macro"
      scope="row"
    >
      <td
        className="cell"
      >
        MU
         
      </td>
      <td
        className="cell positive"
      >
        2136.64
         
      </td>
      <td
        className="cell"
      >
        Macro
         
      </td>
    </tr>
    <tr
      className="cell macro"
      scope="row"
    >
      <td
        className="cell"
      >
        NU
         
      </td>
      <td
        className="cell positive"
      >
        2744.89
         
      </td>
      <td
        className="cell"
      >
        Macro
         
      </td>
    </tr>
    <tr
      className="cell equities"
      scope="row"
    >
      <td
        className="cell"
      >
        OMIKRON
         
      </td>
      <td
        className="cell positive"
      >
        2735.1
         
      </td>
      <td
        className="cell"
      >
        Equities
         
      </td>
    </tr>
    <tr
      className="cell equities"
      scope="row"
    >
      <td
        className="cell"
      >
        SIGMA
         
      </td>
      <td
        className="cell positive"
      >
        1854.19
         
      </td>
      <td
        className="cell"
      >
        Equities
         
      </td>
    </tr>
    <tr
      className="cell macro"
      scope="row"
    >
      <td
        className="cell"
      >
        TAU
         
      </td>
      <td
        className="cell positive"
      >
        2082.71
         
      </td>
      <td
        className="cell"
      >
        Macro
         
      </td>
    </tr>
    <tr
      className="cell equities"
      scope="row"
    >
      <td
        className="cell"
      >
        OMEGA
         
      </td>
      <td
        className="cell positive"
      >
        2306.35
         
      </td>
      <td
        className="cell"
      >
        Equities
         
      </td>
    </tr>
    <tr
      className="cell credit"
      scope="row"
    >
      <td
        className="cell"
      >
        PSI
         
      </td>
      <td
        className="cell positive"
      >
        2997.78
         
      </td>
      <td
        className="cell"
      >
        Credit
         
      </td>
    </tr>
  </tbody>
</table>
`;
